packages/expo-dev-client/android/src/androidTest/java/com/expo/modules/devclient/tests/DevClientDeepLinkTest.kt

```diff
@@ -0,0 +1,53 @@
+package com.expo.modules.devclient.tests
+
+import android.content.Intent
+import android.net.Uri
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import com.expo.modules.devclient.idlingresource.waitUntilViewIsDisplayed
+import com.expo.modules.devclient.koin.DevLauncherKoinTest
+import com.expo.modules.devclient.koin.declareInDevLauncherScope
+import com.expo.modules.devclient.scenarios.DevLauncherBasicScenario
+import expo.modules.devlauncher.launcher.DevLauncherIntentRegistryInterface
+import expo.modules.devlauncher.launcher.DevLauncherPendingIntentListener
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+internal class DevClientDeepLinkTest: DevLauncherKoinTest() {
+  @Test
+  fun checks_if_pending_deep_link_is_displayed() = DevLauncherBasicScenario(koinDeclaration = {
+    val intentRegistry: DevLauncherIntentRegistryInterface = object : DevLauncherIntentRegistryInterface {
+      override var intent: Intent? = null
+
+      override fun subscribe(listener: DevLauncherPendingIntentListener) = Unit
+      override fun unsubscribe(listener: DevLauncherPendingIntentListener) = Unit
+
+      override fun consumePendingIntent(): Intent? = intent
+    }
+
+    intentRegistry.intent = Intent().apply {
+      data = Uri.parse("http://localhost:9876")
+    }
+    @Suppress("RemoveExplicitTypeArguments")
+    declareInDevLauncherScope<DevLauncherIntentRegistryInterface> {
+      intentRegistry
+    }
+  }).setUpAndLaunch {
+    onView(withText("http://localhost:9876")).check(matches(isDisplayed()))
```

Comment:
this is a question rather than a comment, but for my own edification, could you explain why (1) we are sending this intent without any special scheme or the `expo-development-client` hostname, and (2) what is actually being tested here (e.g. what view should display the deep link URL)? 😄

Reply:
1) It doesn't contain any scheme, cause in normal use case it will be pared already. So, the dev-launcher will only save the data which should be pass to the app. 
2) If you send a deep link into the app and it should be handled by the user app - not dev-launcher, but no app is loaded, we show the user short text that informs him about the pending intent. In that test, I've checked if the intent is shown (it will contain the URL of the pending intent).